src/zenml/zen_stores/rest_zen_store.py

```diff
@@ -1424,23 +1457,32 @@ def _get_auth_token(self) -> str:
                 format.
         """
         if self._api_token is None:
-            response = self._handle_response(
-                requests.post(
-                    self.url + API + VERSION_1 + LOGIN,
-                    data={
-                        "username": self.config.username,
-                        "password": self.config.password,
-                    },
-                    verify=self.config.verify_ssl,
-                    timeout=self.config.http_timeout,
-                )
-            )
-            if not isinstance(response, dict) or "access_token" not in response:
-                raise ValueError(
-                    f"Bad API Response. Expected access token dict, got "
-                    f"{type(response)}"
+            # Check if the API token is already stored in the config
+            if self.config.api_token:
+                self._api_token = self.config.api_token
+            # Otherwise, try to get the token from the server
+            else:
+                response = self._handle_response(
+                    requests.post(
+                        self.url + API + VERSION_1 + LOGIN,
+                        data={
+                            "username": self.config.username,
+                            "password": self.config.password,
```

Comment:
same here: I would recommend throwing an exception if the username or password are None, before calling the request.